src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+                }
+                else
+                {
+                    result.IsReadFrom = true;
+                }
+
+                // Override read\writtern information from SymbolUsageInfo
+                if (referenceItem.SymbolUsageInfo.IsWrittenTo() && referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsWrittenTo())
+                {
+                    result.IsReadFrom = false;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = false;
+                }
+
+                var definitionResult = context.GetDefinitionResult(referenceItem.Definition);
+                if (definitionResult != null)
+                {
+                    result.Definition = definitionResult;
+                }
+                else
+                {
+                    System.Diagnostics.Debugger.Break();
+                    // I assumed that definitions are profferred first, but we just received a reference before getting a definition.
+                    // In this case, create a queue of definitions that need to have their Results realized.
+                }
+            }
+            else if (definitionItem != null)
+            {
+                result.IsDefinition = true;
+                if (definitionItem.DisplayableProperties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var containingTypeInfo))
+                {
+                    result.ContainingTypeName = containingTypeInfo;
+                }
+
+                if (definitionItem.DisplayableProperties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var containingMemberInfo))
+                {
+                    result.ContainingMemberName = containingMemberInfo;
+                }
+                result.Kind = "Definition";
+
+                result.Icon = new ImageElement(definitionItem.Tags.GetFirstGlyph().GetImageId());
+
+                // TODO: Use DocumentSpanEntry
+                result.ContextText = new ClassifiedTextElement(excerptResult.ClassifiedSpans.Select(cspan =>
+                    new ClassifiedTextRun(cspan.ClassificationType, sourceText.ToString(cspan.TextSpan))));
+
+                result.ContextHighlightSpan = AsSpan(excerptResult.MappedSpan);
+                // TODO: add definition context and its highlight
+            }
+
+            // Set location so that Editor can navigate to the result
+            var mappedDocumentSpan = await FindUsagesUtilities.TryMapAndGetFirstAsync(documentSpan, sourceText, cancellationToken).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
```

Comment:
when does it not have a value?

Reply:
I don't know of the circumstances.
TryMapAndGetFirstAsync may return null, so I'm programming defensively.